Lib/test/test_tools/test_msgfmt.py

```diff
@@ -240,6 +240,362 @@ def test_strings(self):
                     with self.assertRaises(Exception):
                         msgfmt.make('messages.po', 'messages.mo')
 
+    def test_general_syntax_errors(self):
+        invalid_po_files = (
+            '',
+            '"',
+            '""',
+            '"foo"',
+            # 'msgid',  # invalid but currently accepted
+            'msgstr',
+            'msgid_plural',
+            # 'msgctxt',  # invalid but currently accepted
+            'msgstr',
+            'msgstr[0]',
+            '[0]',
+
+            # unclosed string
+            '''
+msgid "
+msgstr "bar"
+''',
+
+            # unclosed string
+            '''
+msgid "foo
+msgstr "bar"
+''',
+
+            # unclosed string
+            '''
+msgid "foo" "
+msgstr "bar"
+''',
+
+            # unclosed string
+            '''
+msgid "foo"
+"
+msgstr "bar"
+''',
+
+            # illegal backslash
+            '''
+msgid "foo\\"
+"
+msgstr "bar"
+''',
+
+            # msgid with an index
+            '''
+msgid[0] "foo"
+msgstr "bar"
+''',
+
+            # invalid plural index
+            # invalid but currently accepted
+#             '''
+# msgid "foo"
```

Comment:
I do not think that it is worth to add such large blocks of the commented out code. It only increases the diff, in this PR, and in the future ones.

Reply:
Removed, I'll add them back (uncommented) when I fix the parser